snuba/utils/streams/streaming.py

```diff
@@ -132,7 +137,29 @@ def __len__(self) -> int:
         return len(self.__items)
 
     def __getitem__(self, index: int) -> Message[TPayload]:
+        """
+        Get a message in this batch by its index.
+
+        The message returned by this method is effectively a copy of the
+        original message within this batch, and may be safely passed
+        around without requiring any special accomodation to keep the shared
+        block open or free from conflicting updates.
+        """
         data, buffers = self.__items[index]
+        # The buffers read from the shared memory block are converted to
+        # ``bytes`` rather than being forwarded as ``memoryview`` for two
+        # reasons. First, true buffer support protocol is still pretty rare (at
+        # writing, it is not supported by either standard library ``json`` or
+        # ``rapidjson``, nor by the Confluent Kafka producer), so we'd be
+        # copying these values at a later stage anyway. Second, copying these
+        # values from the shared memory block (rather than referencing location
+        # within it) means that we do not have to ensure that the shared memory
+        # block is not recycled during the life of one of these ``Message``
+        # instances. If the shared memory block was reused for a different
+        # batch while one of the ``Message`` instances returned by this method
+        # was still "alive" in a different part of the processing pipeline, the
+        # contents of the message are liable to be corrupted (at best --
```

Comment:
now, I may be wrong on this.... isn't this `are` supposed to be `would be` ?

Reply:
Yeah, that would be clearer. Will update.